pkg/apis/calico/validation/networkconfig_test.go

```diff
@@ -0,0 +1,129 @@
+						corev1.ResourceMemory: resource.MustParse("128Mi"),
+					},
+					Typha: &corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("50m"),
+						corev1.ResourceMemory: resource.MustParse("64Mi"),
+					},
+				},
+			},
+		}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should fail with a static autoscaling mode and invalid resource name", &apiscalico.NetworkConfig{
+			AutoScaling: &apiscalico.AutoScaling{
+				Mode: apiscalico.AutoscalingModeStatic,
+				Resources: &apiscalico.StaticResources{
+					Node: &corev1.ResourceList{
+						"invalid": resource.MustParse("128Mi"),
+					},
+					Typha: &corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("50m"),
+						corev1.ResourceMemory: resource.MustParse("64Mi"),
+					},
+				},
+			},
+		}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Field": Equal("config.autoScaling.resources.node.invalid"),
+			})))),
+		Entry("should return error with invalid IPIP mode", &apiscalico.NetworkConfig{IPIP: ptr.To(apiscalico.PoolMode("invalid"))}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipip")})))),
+		Entry("should succeed with valid IPIP mode", &apiscalico.NetworkConfig{IPIP: ptr.To(apiscalico.Always)}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=cali1234")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method with interface and regex", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=^cali.*")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method with interface", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=cali1234")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method with a valid cidr", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("cidr=192.168.0.0/16")}, field.NewPath("config"),
+			BeEmpty()),
+
+		Entry("should return error with empty IP autodetection method", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+
+		Entry("should return error with invalid IP autodetection method format", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("invalid-format")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+		Entry("should return error with invalid IP autodetection method regex", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=invalid[regex")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipAutoDetectionMethod")})))),
+
```

Comment:
I am confused by the seemingly random use of empty lines. Could you please remove them for consistency?